Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -86,4 +91,96 @@ namespace ROS2
         return joints;
     }
 
+    AZStd::unordered_set<AZStd::string> Utils::GetMeshesFilenames(const urdf::LinkConstSharedPtr& root_link, bool visual, bool colliders)
+    {
+        AZStd::unordered_set<AZStd::string> filenames;
+        const auto getFnFromGeometry = [](const urdf::GeometrySharedPtr& geometry)
+        {
+            if (geometry->type == urdf::Geometry::MESH)
+            {
+                auto pMesh = std::dynamic_pointer_cast<urdf::Mesh>(geometry);
+                if (!pMesh)
+                {
+                    return AZStd::string{};
+                }
+                return AZStd::string(pMesh->filename.c_str(), pMesh->filename.size());
+            }
+            return AZStd::string{};
+        };
+
+        const auto processLink = [&](const urdf::Link& link)
+        {
+            if (visual)
+            {
+                for (auto& p : link.visual_array)
+                {
+                    const auto fn = getFnFromGeometry(p->geometry);
+                    if (!fn.empty())
+                    {
+                        filenames.insert(fn);
+                    }
+                }
+            }
+            if (colliders)
+            {
+                for (auto& p : link.collision_array)
+                {
+                    const auto fn = getFnFromGeometry(p->geometry);
+                    if (!fn.empty())
+                    {
+                        filenames.insert(fn);
+                    }
+                }
+            }
+        };
+
+        std::function<void(const std::vector<urdf::LinkSharedPtr>&)> link_visitor =
+            [&](const std::vector<urdf::LinkSharedPtr>& child_links) -> void
+        {
+            for (auto child_link : child_links)
+            {
+                processLink(*child_link);
+                link_visitor(child_link->child_links);
+            }
+        };
+        processLink(*root_link);
+        link_visitor(root_link->child_links);
```

Comment:
we can simply call the linkVisitor (note the naming convention) for the root ( as a single-element vector)
Also, rename child_links in the lambda above to "links" since this is the level where they are actually processed

Reply:
I think is suboptimal solution, since there is a little mess in URDF lib with std::shared_ptr<const ...>